src/main/java/uk/gov/hmcts/ccd/domain/service/startevent/DefaultStartEventOperation.java

```diff
@@ -64,41 +76,70 @@ public StartEventTrigger triggerStartForCaseType(final String uid,
 
         callbackInvoker.invokeAboutToStartCallback(eventTrigger, caseType, caseDetails, ignoreWarning);
 
         return buildStartEventTrigger(eventTriggerId, eventToken, caseDetails);
     }
 
     @Override
-    public StartEventTrigger triggerStartForCase(final String uid,
-                                                 final String jurisdictionId,
-                                                 final String caseTypeId,
-                                                 final String caseReference,
-                                                 final String eventTriggerId,
-                                                 final Boolean ignoreWarning) {
+    public StartEventTrigger triggerStartForDraft(final String uid,
+                                                  final String jurisdictionId,
+                                                  final String caseTypeId,
+                                                  final String draftReference,
+                                                  final String eventTriggerId,
+                                                  final Boolean ignoreWarning) {
+        return buildStartEventTrigger(uid,
+                                      jurisdictionId,
+                                      caseTypeId,
+                                      eventTriggerId,
+                                      ignoreWarning,
+            () -> getDraftDetails(jurisdictionId, draftReference));
+    }
 
+    private StartEventTrigger buildStartEventTrigger(final String uid,
+                                                     final String jurisdictionId,
+                                                     final String caseTypeId,
+                                                     final String eventTriggerId,
+                                                     final Boolean ignoreWarning,
+                                                     final Supplier<CaseDetails> caseDetailsSupplier) {
         final CaseType caseType = getCaseType(caseTypeId);
         final CaseEvent eventTrigger = getEventTrigger(caseTypeId, eventTriggerId, caseType);
 
         validateJurisdiction(jurisdictionId, caseTypeId, caseType);
 
-        final CaseDetails caseDetails = getCaseDetails(jurisdictionId, caseTypeId, caseReference);
+        final CaseDetails caseDetails = caseDetailsSupplier.get();
 
-        validateEventTrigger(() -> !eventTriggerService.isPreStateValid(caseDetails.getState(), eventTrigger));
+        validateEventTrigger(() -> !eventTriggerService.isPreStateEmpty(eventTrigger));
 
-        final String eventToken = eventTokenService.generateToken(uid, caseDetails, eventTrigger,
-            caseType.getJurisdiction(), caseType);
+        // TODO: we may need to take care of drafts that are saved for existing case so token needs to include the relevant draft payload
```

Comment:
That's an interesting point... I'm not sure how the drafts would fit in the existing event token flow.

Reply:
Let's discuss once we actually need that. The comment was just to make us aware.